bgpd/bgp_nexthop.c

```diff
@@ -753,6 +760,22 @@ static void bgp_show_nexthop_paths(struct vty *vty, struct bgp *bgp,
 		safi = table->safi;
 		bgp_path = table->bgp;
 
+		if (json) {
+			json_path = json_object_new_object();
+			json_object_string_add(json_path, "afi", afi2str(afi));
+			json_object_string_add(json_path, "safi",
+					       safi2str(safi));
+			if (dest->pdest)
+				json_object_string_addf(
+					json_path, "rd", "%pRD",
+					(struct prefix_rd *)bgp_dest_get_prefix(
```

Comment:
Please use prefix_rd2str() to print RD. Example: "prefix_rd2str((struct prefix_rd *)bgp_dest_get_prefix(dest->pdest), buf1, sizeof(buf1));"

Reply:
Why should? `%pRD` does the same.

Reply:
I applied this diff to my workspace and this is what I see:

JSON output for "show bgp nexthop 6.0.0.28"
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
'''
"lastUpdate":1666743111,
      "paths":[
        {
          "afi":"l2vpn",
          "safi":"evpn",
          "rd":"0x555a4bde3e70RD",
          "name":"VRF default",
          "flags":1040
        },
'''

regular vtysh output for "show bgp nexthop 6.0.0.28"
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Paths:
    3/5 [3]:[6.0.0.28]/352 RD 6.0.0.28:7 VRF default flags 0x410
    3/5 [2]:[00:02:00:00:00:0b]:[fe80::202:ff:fe00:b]/352 RD 6.0.0.28:7 VRF default flags 0x410


regular Vtysh output displays RD string,  but JSON output displays the address of RD.